### PR TITLE
Change low planet density

### PR DESCRIPTION
**Pull request by LGM-Doyle**
This changes the percentage change of systems with no planet in a slot to 85% from 77% for the low density case.  The medium density case is 70%.  The previous low density case was 77%.  In play, I couldn't perceive the difference between 70% and 77%.  Perhaps even 90% is playable and fun.

I will post this to the website for comments as it is a game play change.

### CODE REVIEW

File: Empire/Empire.cpp [hunk 1]
```diff
@@ -2930,7 +2930,7 @@ void Empire::CheckProductionProgress() {
 
         // consume the item's special and meter consumption
         std::map<std::string, std::map<int, float> > sc = elem.item.CompletionSpecialConsumption(elem.location);
-        for (std::map<std::string, std::map<int, float>>::iterator sc_it = sc.begin(); sc_it != sc.end(); ++sc_it) {
+        for (std::map<std::string, std::map<int, float> >::iterator sc_it = sc.begin(); sc_it != sc.end(); ++sc_it) {
```
Comment (geoffthemedio) [inline Empire/Empire.cpp:2933]: maybe you should submit this as a separate pull request, rather than including it in multiple unrelated ones?

**Comment by LGM-Doyle** [comment]
Adrian fixed in 7f445dd80c40e31be9a3240c461f9d7c37613d8d.

**Comment by Vezzra** [comment]
I wonder how much this change will affect game balance in the end, how much testing we're going to need to actually get an accurate feeling for this, and consequently if we really should include that in 0.4.6, or better postpone it for afterwards.

Please give your opinions on that.

I'm keeping the 0.4.6 milestone for now.

**Comment by LGM-Doyle** [comment]
I played 5 games.  The AIs expanded, formed empires and were conquered as normal.  The change was a fun change that moved the galaxy in the direction of resource sparse, giving empires a reason for conflict.

user0169 clearly articulated some good reasons to include options to create galaxies like this in:
http://freeorion.org/forum/viewtopic.php?f=28&t=10049&sid=f2ae0d4e0976ea001bc28b7babcfc13c

I'm also linking in the forum discussion .
http://freeorion.org/forum/viewtopic.php?f=6&t=10008

@MatGB  wanted a broader code change to exclude the creation of stars with no planets/asteroids at all.

You suggested increasing the planet density in the medium galaxies to differentiate medium from low.  I think what I was hoping for and user0169 as well is a low resource poor galaxy with long distances between the good stuff.  This makes all the research and colonization choices new and interesting.  This PR does not go far enough in my opinion, but it also does not break the AI. 

I would say include this PR.  Then add a future change items to prevent empty stars as @MatGB  requested and allow more resource scarcity with the provision that the AI be tested with respect to its responsiveness to scarcity in its research and colonization choices.

**Comment by Vezzra** [comment]
Hm, I see. Well, as this change is actually a very simple one that can be easily reversed should it turn out to not work well, I'm inclined to merge, so it will get more testing.

Unless anyone objects, I'm going to merge that before I produce the upcoming test builds (hopefully I don't forget... :wink:).
